File: AcademicPlanBackend/src/main/java/com/bntu/fitr/poit/zholudev/diplom/entity/CourseProjectTask.java
package com.bntu.fitr.poit.zholudev.diplom.entity;

import javax.persistence.*;
import javax.persistence.Entity;

@Entity
@Table(name = "course_project_tasks")
public class CourseProjectTask {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String text;

    public CourseProjectTask() {
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }
}

File: AcademicPlanFrontend/src/app/entity/course-project-task.ts
export class CourseProjectTask {
  id: number;
  text: string;

  static clone(courseProjectTask: CourseProjectTask): CourseProjectTask {
    let clonedCloseProjectTask: CourseProjectTask = new CourseProjectTask();
    clonedCloseProjectTask.id = courseProjectTask.id;
    clonedCloseProjectTask.text = courseProjectTask.text;
    return clonedCloseProjectTask;
  }
}
